app/models/moment.rb

```diff
@@ -38,6 +38,15 @@ class Moment < ApplicationRecord
   validates :userid, :name, :why, presence: true
   validates :why, length: { minimum: 1, maximum: 2000 }
   validates :fix, length: { maximum: 2000 }
+  validates :secret_share_expires_at,
+            presence: true, if: :secret_share_identifier?
+
+  scope :find_secret_share!, lambda { |identifier|
+    find_by!(
+      'secret_share_expires_at > NOW()',
```

Comment:
Is NOW() equivalent to Time.now?

Reply:
Welcome to SQL

Reply:
Technically, this could be done with Arel but I thought it was a little early to scare them with that.

Would be something like:
```
find_by!(
  arel_table[: secret_share_expires_at].gt(Time.zone.now),
  secret_share_identifier: identifier
)
```

Reply:
Ah gotcha, thanks for explaining! I haven't touched pure SQL in a while :P
What's the advantage of using Arel @HashNotAdam (other than not having to write SQL)?

Reply:
Here... my (ever humble) opinion is nothing.

It can be used programmatically, though, so if there was a situation where you wanted to pass around objects like you do with standard ActiveRecord elements that might take variables, you can have different methods tagging on pieces of your query without worrying about the order you are concatenating your string or replacing variables in a string.

It also means you don't need to worry about DB specific syntax. `NOW()` is extremely common but it is _really_ common for syntax between DBs to be different and, like with AR, using Arel means you can change from, say, PostgreSQL to MySQL without rewriting code.